libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -592,7 +592,7 @@ string YulUtilFunctions::nextArrayElementFunction(ArrayType const& _type)
 		if (_type.location() == DataLocation::Memory)
 			templ("advance", "0x20");
 		else if (_type.location() == DataLocation::Storage)
-			templ("advance", "1");
```

Comment:
Please also add an assertion that it is `>= 1`.

Reply:
There is already `solAssert(_type.baseType()->storageBytes() > 16, "")` in line number 583. Do we want to assert again?

Reply:
(I mean, `storageBytes() > 16` somewhat implies `storageSize()` is not zero?)

Reply:
Better safe than sorry! :)

Reply:
Ok. I also add an assertion for `calldataEncodedSize()`

Reply:
I hope `solAssert(size >= 32 && size % 32 == 0, "");` makes sense for calldata types. Looks like they always pad to word except `bytes` and `strings`.

Reply:
Yep, looks good!